### ISSUE TITLE
Improvement: Merge IsEmpty(BlockSyntax) implementations

### ISSUE THREAD

**Issue by martin-strecker-sonarsource** [SonarSource/sonar-dotnet#5875]
### Description

There is more than one implementation for checking if an SyntaxBlock is empty. One is in [EmptyMethod.cs](https://github.com/SonarSource/sonar-dotnet/blob/ccf3ce1460ecf8d641ecc793164dd8225367bf62/analyzers/src/SonarAnalyzer.CSharp/Rules/EmptyMethod.cs#L86-L87) and another one in [EmptyNestedBlock.cs](https://github.com/SonarSource/sonar-dotnet/blob/ccf3ce1460ecf8d641ecc793164dd8225367bf62/analyzers/src/SonarAnalyzer.CSharp/Rules/EmptyNestedBlock.cs#L66-L67) and there might be others too. The implementation in EmptyNestedBlock is superior to the one in EmptyMethod even after #5855 is merged.

Both implementations should be merged into an extension methods and tests for the extension method should be written.

See also https://github.com/SonarSource/sonar-dotnet/pull/5855/files#r924674589

Blocked by #5855
